### PR TITLE
kinesis put records

### PR DESCRIPTION
**Pull request by andreineculau**
wip. needs tests
don't know if it's worth trying to make this and firehose.putRecords abstract

### CODE REVIEW

**Comment by IanSavchenko** [comment]
Abstract in a sense to be reused between these two? I don't think it should, it's just a coincidence that they have same (or just similar?) API and there might be more differences later.

**Comment by andreineculau** [comment]
@IanSavchenko refreshed the PR, just in case you're ok merging it

File: src/kinesis.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  _.forEach(records, function(record) {
+    let Data = JSON.stringify(record);
+    let dataLength = Buffer.byteLength(JSON.stringify({
+      Data: record,
+      ExplicitHashKey,
+      PartitionKey
+    }));
+
+    if (dataLength > limits.recordByteSize) {
+      ctx.log.error(`Skipping record larger than ${limits.recordByteSize / 1024} KB: \
```
Comment (IanSavchenko) [inline src/kinesis.js]: I'm questioning now this approach where a lib function almost silently fails to send records. I know we have it in the firehose too, but still. I would say it's up to the app to decide how to deal with such records.

Should we maybe just expose a helper function that checks a record size and allows a client to decide what to do with non-conforming records?

Comment (andreineculau) [reply src/kinesis.js]: Sounds good! But as part of another PR

**Pull request merged**